### DISCUSSION TITLE
UDP data source & sink

### DISCUSSION THREAD

**Discussion by amarnathpv** [vectordotdev/vector#14982] [Q&A] [answered]
Greetings!

Is there source & sink to support UDP data stream and forward to other UDP consumers?

With that, to retain the source IP as spoofing the data - this is similar to samplicate tool.

Thanks.

**Comment by jszwedko** [comment, marked as answer]
Hi @amarnathpv !

You should be able to use the `socket` source and sink for this. They both support UDP. The sink doesn't support spoofing the IP address to use the origin IP though. That would need to be a feature request if you want to open one.

**Comment by amarnathpv** [comment]
Thanks for your response @jszwedko.
